### PR TITLE
27 frontend fix sidebar layout

### PR DESCRIPTION
**Pull request by Mitashi-11**
This branch adds functionality to the star and favorite buttons.

### CODE REVIEW

File: src/frontend/src/Components/EditBar.css [hunk 1]
```diff
@@ -12,7 +11,7 @@
     margin-left: 20px;
 }
 .buttons{
-    margin-left: auto;
+    margin-left: 0;
```
Comment (jluna28) [inline src/frontend/src/Components/EditBar.css:14]: remove since not needed

File: src/frontend/src/Components/EditBar.css [hunk 2]
```diff
@@ -12,7 +11,7 @@
...
     margin-right: 0;
```
Comment (jluna28) [inline src/frontend/src/Components/EditBar.css:15]: remove since not needed

File: src/frontend/src/Components/SideBar.tsx [hunk 3]
```diff
@@ -1,31 +1,150 @@
-import React from 'react';
+import React, {useState} from 'react';
 import SideBarItem from "./SideBarItem";
 import './SideBar.css';
+import AddBoxOutlinedIcon from '@mui/icons-material/AddBoxOutlined';
+import IndeterminateCheckBoxOutlinedIcon from '@mui/icons-material/IndeterminateCheckBoxOutlined';
+import {IconButton, Stack, Tooltip} from "@mui/material";
+
+interface flowchartProps{
```
Comment (jluna28) [inline src/frontend/src/Components/SideBar.tsx:8]: should be with the other interfaces

File: src/frontend/src/Components/SideBarItem.tsx [hunk 4]
```diff
@@ -1,37 +1,51 @@
...
 import StarBorderIcon from '@mui/icons-material/StarBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
 import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
-
 import "./SideBarItem.css";
 
-function SideBarItem (props: {id : bigint,  name : string}) : JSX.Element {
-    let clickEvent = () => {
-        alert("you clicked on " + props.name  + " with id: " + props.id)
+interface SideBarItemProps {
```
Comment (jluna28) [inline src/frontend/src/Components/SideBarItem.tsx:9]: should be with other interfaces

File: src/frontend/src/Components/SideBarItem.tsx [hunk 5]
```diff
@@ -1,37 +1,51 @@
...
+        }
+    };
+
     return (
         <div className="sideBarItem">
-            {/*<Stack direction ="row" alignItems="center">*/}
-            {/*    <Typography noWrap onClick={() => clickEvent()} sx={{ minWidth: 0, maxWidth: 120}} >{props.name}</Typography>*/}
-            <span className="barItemTitle" onClick={() => clickEvent()}>
-                <p>{props.name}</p>
+            <span className="barItemTitle" onClick={clickEvent}>
```
Comment (jluna28) [inline src/frontend/src/Components/SideBarItem.tsx:35]: check if this is needed

**Review by jluna28 — CHANGES_REQUESTED** [review]

File: src/frontend/src/Components/SideBarItem.tsx (hunk 5 above)
Comment (Mitashi-11) [reply src/frontend/src/Components/SideBarItem.tsx:35]: Yes, the text class overrides some of the inbuilt style of the <p>. The barItemTitle sets the layout the buttons w.r.t to the text.

**Review by jluna28 — APPROVED** [review]

**Pull request merged**